wsi/drm/drmbuffer.h

```diff
@@ -93,16 +93,16 @@ class DrmBuffer : public OverlayBuffer {
 
   void Dump() override;
 
+  void SaveMetaDataBufferObject(const HwcBuffer& bo) { mtdt = bo; };
+
  private:
   void Initialize(const HwcBuffer& bo);
   uint32_t width_ = 0;
   uint32_t height_ = 0;
   uint32_t format_ = 0;
   uint32_t tiling_mode_ = 0;
   uint32_t frame_buffer_format_ = 0;
-  uint32_t pitches_[4];
-  uint32_t offsets_[4];
-  uint32_t gem_handles_[4];
+  HwcBuffer mtdt; /* metadata buffer object */
```

Comment:
hmm no, why cannot we just retrieve the infor from HWcbuffer which is part of the handle

Reply:
ya

Reply:
We still need to save the reference to 'HWCBuffer bo' because it is passed in only thru DrmBuffer::Initialize() and being referenced elsewhere.

Reply:
hmm, hwchandle has instance for its bo, which we should use everywhere. in initialize we should make sure that gets populated correctly and use that everywhere as needed.